File: example_services/scala/finatra/example/example-service/src/main/scala/com/me/example/handlers/PostMessageHandler.scala
package com.me.example.handlers

import javax.inject.Inject

import com.me.example.domain.http.PostMessageRequest
import com.me.example.persistence.MessageDao
import com.twitter.inject.Logging
import com.twitter.util.Future

class PostMessageHandler @Inject()(
  dao: MessageDao,
  request: PostMessageRequest
) extends RequestHandler
  with Logging {

  override def handle(): Future[Any] = {
    info(s"Processing: $request")
    dao.set(request.id, request.message).map { _ => () }
  }
}

File: example_services/java/finatra/example/example-service/src/main/java/com/me/example/handlers/PostMessageHandler.java
package com.me.example.handlers;

import com.me.example.persistence.MessageDao;
import com.twitter.finagle.http.Request;


public class PostMessageHandler implements RequestHandler {

    private final MessageDao messageDao;

    public PostMessageHandler(MessageDao messageDao) {
        this.messageDao = messageDao;
    }

    @Override
    public Object handle(Request request) {
        String key = request.getParam("id");
        messageDao.set(key, request.contentString());
        return request.contentString();
    }
}
